examples/v1beta1/pipelineruns/pipelinerun.yaml

```diff
@@ -26,82 +14,165 @@ roleRef:
+          rm -rf "$CHECKOUT_DIR"/..?*
+        fi
+      }
+      if [[ "$(params.deleteExisting)" == "true" ]] ; then
+        cleandir
+      fi
+      /ko-app/git-init \
+        -url "$(params.url)" \
+        -revision "$(params.revision)" \
+        -path "$CHECKOUT_DIR" \
+        -sslVerify="$(params.sslVerify)" \
+        -submodules="$(params.submodules)" \
+        -depth="$(params.depth)"
+      cd "$CHECKOUT_DIR"
+      RESULT_SHA="$(git rev-parse HEAD | tr -d '\n')"
+      EXIT_CODE="$?"
+      if [ "$EXIT_CODE" != 0 ]
+      then
+        exit $EXIT_CODE
+      fi
+      # Make sure we don't add a trailing newline to the result!
+      echo -n "$RESULT_SHA" > $(results.commit.path)
+---
+# Copied from https://github.com/tektoncd/catalog/blob/v1beta1/kaniko/kaniko.yaml
+# with a few fixes that will be port over in https://github.com/tektoncd/catalog/pull/291
+# Post #1839 we can refer to the remote Task in a registry or post #2298 in git directly
 apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
-  name: build-push
+  name: kaniko
 spec:
+  workspaces:
+  - name: source
   params:
-  - name: pathToDockerFile
-    description: The path to the dockerfile to build
-    default: $(resources.inputs.docker-source.path)/Dockerfile
-  - name: pathToContext
-    description: The build context used by Kaniko (https://github.com/GoogleContainerTools/kaniko#kaniko-build-contexts)
-    default: $(resources.inputs.docker-source.path)
-  resources:
-    inputs:
-    - name: docker-source
-      type: git
-    outputs:
-    - name: builtImage
-      type: image
+  - name: IMAGE
+    description: Name (reference) of the image to build.
+  - name: DOCKERFILE
+    description: Path to the Dockerfile to build.
+    default: ./Dockerfile
+  - name: CONTEXT
+    description: The build context used by Kaniko.
+    default: ./
+  - name: EXTRA_ARGS
+    default: ""
+  - name: BUILDER_IMAGE
+    description: The image on which builds will run
+    default: gcr.io/kaniko-project/executor:latest
+  results:
+  - name: IMAGE_DIGEST
+    description: Digest of the image just built.
   steps:
   - name: build-and-push
-    image: gcr.io/kaniko-project/executor:v0.17.1
+    workingDir: $(workspaces.source.path)
+    image: $(params.BUILDER_IMAGE)
     # specifying DOCKER_CONFIG is required to allow kaniko to detect docker credential
+    # https://github.com/tektoncd/pipeline/pull/706
     env:
-    - name: "DOCKER_CONFIG"
-      value: "/tekton/home/.docker/"
+    - name: DOCKER_CONFIG
+      value: /tekton/home/.docker
     command:
     - /kaniko/executor
+    - $(params.EXTRA_ARGS)
+    - --dockerfile=$(params.DOCKERFILE)
+    - --context=$(workspaces.source.path)/$(params.CONTEXT)  # The user does not need to care the workspace and the source.
+    - --destination=$(params.IMAGE)
+    - --oci-layout-path=$(workspaces.source.path)/$(params.CONTEXT)/image-digest
+    securityContext:
+      runAsUser: 0
```

Comment:
Comment here?

Reply:
This is copied directly from the catalog - I'll copy the comment from the commit message (https://github.com/tektoncd/catalog/commit/0a8b65343b0c8f7409366141c741ec6296764cd9) and if i'm feeling very ambitious ill open a PR to fix this in catalog also